src/roomHandlers/fishingRoomHandler.ts

```diff
@@ -0,0 +1,129 @@
+import { RoomHandler } from './roomHandler';
+
+import {
+  Command,
+  CommandHandlersByCommandByPlayerStates,
+  DEFAULT_COMMAND,
+  NarrowedRoomState,
+  NullableCommandHandler,
+  StateDescriptionAccessor,
+} from '../utils/types';
+
+const roomType = 'fishingRoom';
+type TRoomType = typeof roomType;
+type TRoomState = NarrowedRoomState<TRoomType>;
+type TStateDescriptionAccessor = StateDescriptionAccessor<TRoomState>;
+type TCommandHandlersByCommandByPlayerStates =
+  CommandHandlersByCommandByPlayerStates<TRoomType>;
+type TNullableCommandHandler = NullableCommandHandler<TRoomType>;
+
+const stateDescriptionAccessor: TStateDescriptionAccessor = {
+  AtEntrance: ({ fishCaught, isRodBroken }) => {
+    const poleText = isRodBroken ? 'broken' : 'working';
+
+    return {
+      playerStateDescription: `You are on a dock with a ${poleText} fishing pole. You have caught ${fishCaught} fish so far!`,
+      availableCommands: ['fish', 'leave'],
+    };
+  },
+  Fishing: {
+    playerStateDescription: 'Hopefully the fish are hungry today.',
+    availableCommands: ['stop', 'fish'],
+  },
+  Caught: {
+    playerStateDescription: 'Congratulations!!! You caught a fish!!!',
+    availableCommands: ['goToEntrance'],
+  },
+  BrokenLine: {
+    playerStateDescription:
+      'You have a broken fishing line and the fish got away.',
+    availableCommands: ['stop', 'fix'],
+  },
+};
+
+const commandHandlersByCommandByPlayerState: TCommandHandlersByCommandByPlayerStates =
+  {
+    AtEntrance: {
+      fish: (roomState) => ({
+        commandDescription: 'You start to fish.',
+        roomState: {
+          ...roomState,
+          playerState:
+            roomState.isRodBroken === true ? 'BrokenLine' : 'Fishing',
+        },
+      }),
+      leave: () => ({
+        commandDescription: 'You say goodbye to the fish and leave.',
```

Comment:
nit: Your doc and code are out of sync. I don't really care because there isn't an automated way to check that